src/main/java/org/candlepin/resource/OwnerResource.java

```diff
@@ -1995,13 +1997,14 @@ public ProductDTO addContent(String ownerKey, String productId, String contentId
     public ProductDTO removeBatchContent(String ownerKey, String productId, List<String> contentIds) {
         Owner owner = this.getOwnerByKey(ownerKey);
 
-        // Get the matching owner_product & lock it while we are doing the update for this org
+        // Lock the owner_product while we are doing the update for this org
         // This is done in order to prevent collisions in updates on different parts of the product
-        OwnerProduct ownerProduct = ownerProductCurator.getOwnerProductByProductId(owner, productId);
-        ownerProductCurator.lock(ownerProduct);
-        ownerProductCurator.refresh(ownerProduct);
-
-        Product product = ownerProduct.getProduct();
+        if (!this.ownerProductCurator.lockOwnerProduct(owner, productId, LockModeType.PESSIMISTIC_WRITE)) {
+            throw new NotFoundException(
```

Comment:
This statement can fit on a single line

Reply:
I didn't repeatedly tag each instance, but all of them could be addressed the same way.

Normally I wouldn't hold up a PR for this, but since we need to re-run tests anyway to see if the SonarQube one will play nice, this minor correction could serve to do that.